apps/crn-server/test/data-providers/contentful/research-output.data-provider.test.ts

```diff
@@ -0,0 +1,1541 @@
+        {
+          ...getResearchOutputCreateDataObject(),
+          createdBy: 'user-0',
+        },
+        { publish: false },
+      );
+      const args = environmentMock.createEntry.mock.lastCall?.[1];
+      expect(args!.fields).toMatchObject({
+        createdBy: {
+          'en-US': {
+            sys: {
+              type: 'Link',
+              linkType: 'Entry',
+              id: 'user-0',
+            },
+          },
+        },
+        updatedBy: {
+          'en-US': {
+            sys: {
+              type: 'Link',
+              linkType: 'Entry',
+              id: 'user-0',
+            },
+          },
+        },
+      });
+    });
+
+    test('sets createdDate and lastUpdatedPartial to the current timestamp', async () => {
+      await researchOutputDataProvider.create(
+        getResearchOutputCreateDataObject(),
+        { publish: false },
+      );
+      const args = environmentMock.createEntry.mock.lastCall?.[1];
+      expect(args!.fields).toMatchObject({
+        createdDate: {
+          'en-US': new Date().toISOString(),
+        },
+        lastUpdatedPartial: {
+          'en-US': new Date().toISOString(),
+        },
+      });
+    });
+
+    test('sets addedDate to the current timestamp if publishing the output', async () => {
+      await researchOutputDataProvider.create(
+        getResearchOutputCreateDataObject(),
+        { publish: true },
+      );
+      const args = environmentMock.createEntry.mock.lastCall?.[1];
+      expect(args!.fields).toMatchObject({
+        addedDate: {
+          'en-US': new Date().toISOString(),
+        },
+      });
+    });
+
+    test('sets addedDate to null if not publishing the output', async () => {
+      await researchOutputDataProvider.create(
+        getResearchOutputCreateDataObject(),
+        { publish: false },
+      );
+      const args = environmentMock.createEntry.mock.lastCall?.[1];
+      expect(args!.fields).toMatchObject({
+        addedDate: { 'en-US': null },
+      });
+    });
+  });
+  describe('update', () => {
+    let publish: jest.MockedFunction<() => Promise<Entry>> = jest.fn();
+    let entry: Entry;
+
+    beforeEach(() => {
+      entry = {
+        sys: {
+          publishedVersion: 1,
+        },
+        fields: {},
+        publish,
+      } as unknown as Entry;
+
+      contentfulGraphqlClientMock.request.mockResolvedValue({
+        researchOutputs: {
+          sys: {
+            publishedVersion: 1,
+          },
+        },
+      });
+
+      environmentMock.getEntry.mockResolvedValue(entry);
+      publish.mockResolvedValue(entry);
+
+      const mockPatch = patch as jest.MockedFunction<typeof patch>;
+      mockPatch.mockResolvedValue(entry);
+    });
+
+    test('can update an output', async () => {
+      await researchOutputDataProvider.update(
+        'a',
+        getResearchOutputUpdateDataObject(),
+        { publish: false },
+      );
+
+      expect(patch).toHaveBeenCalled();
```

Comment:
you might want to test what it was called with cause i don't think the 'a' argument is being tested anywhere?